vllm/worker/hpu_model_runner.py

```diff
@@ -97,7 +98,10 @@ def subtuple(obj: object,
     if to_override is None:
         to_override = {}
     fields = set(to_copy) | set(to_override.keys())
-    values = {f: to_override.get(f, getattr(obj, f)) for f in fields}
+    if type(obj) is dict:
+        values = {key: obj[key] for key in fields if key in obj}
```

Comment:
Is this method called often or sizes of object fields huge?

keys_in_obj = [key for key in fields if key in obj]
values = {key: obj[key] for key in keys_in_obj}

I think (based on obj size 1000 and fields 100 in timeit) prefiltering may decrease number of total iterations depending on the size of the obj.